test/foundry/TokenTransferrer.t.sol

```diff
@@ -0,0 +1,274 @@
+                itemType,
+                address(noCodeTokenAddress),
+                address(alice),
+                address(bob),
+                0,
+                1
+            );
+
+            vm.expectRevert(
+                abi.encodeWithSignature(
+                    "NoContract(address)",
+                    noCodeTokenAddress
+                )
+            );
+            test(
+                this.execute,
+                Context(referenceConduit, noCodeTransfer, noCodeBatchTransfer)
+            );
+            vm.expectRevert(
+                abi.encodeWithSignature(
+                    "NoContract(address)",
+                    noCodeTokenAddress
+                )
+            );
+            test(
+                this.execute,
+                Context(conduit, noCodeTransfer, noCodeBatchTransfer)
+            );
+        }
+
+        // Test the 1155 batch transfer no code revert.
+        noCodeBatchTransfer[0] = ConduitBatch1155Transfer(
+            address(noCodeTokenAddress),
+            address(alice),
+            address(bob),
+            new uint256[](0),
+            new uint256[](0)
+        );
+
+        vm.expectRevert(
+            abi.encodeWithSignature("NoContract(address)", noCodeTokenAddress)
+        );
+        test(
+            this.executeBatch,
+            Context(referenceConduit, noCodeTransfer, noCodeBatchTransfer)
+        );
+        vm.expectRevert(
+            abi.encodeWithSignature("NoContract(address)", noCodeTokenAddress)
+        );
+        test(
+            this.executeBatch,
+            Context(conduit, noCodeTransfer, noCodeBatchTransfer)
+        );
+
+        // Test the generic failure case where the token contract reverts.
+        ConduitTransfer[] memory revertTransfer;
+        revertTransfer = new ConduitTransfer[](1);
+
+        TestERC20Revert tokenRevert;
+        tokenRevert = new TestERC20Revert();
+        vm.label(address(tokenRevert), "tokenRevert");
+
+        revertTransfer[0] = ConduitTransfer(
+            ConduitItemType.ERC20,
+            address(tokenRevert),
+            address(alice),
+            address(bob),
+            0,
+            1
+        );
+
+        // I thought I'd be getting the TokenTransferGenericFailure for both
+        // this test and the notOkTransfer test below, but I'm not.
+        // vm.expectRevert(
+        //     abi.encodeWithSignature(
+        //         "TokenTransferGenericFailure(address, address, address, uint256, uint256)",
+        //         address(tokenRevert),
+        //         address(alice),
+        //         address(bob),
+        //         0,
+        //         1
+        //     )
+        // );
+        // No bubbling up in the reference contract.
+        vm.expectRevert();
+        test(
+            this.execute,
+            Context(referenceConduit, revertTransfer, noCodeBatchTransfer)
+        );
+        vm.expectRevert("Some ERC20 revert message");
```

Comment:
I'm also a little confused why this is working... I feel like the `test` wrapper theoretically should just revert with `DifferentialTestAssertionFailed()` when it reverts, but also I know that's not always the case 🤔  Maybe because it's a string revert?